File: src/main/java/editor/database/version/UpdateFrequency.java
package editor.database.version;

/**
 * This enum specifies the frequency at which a database udpate is desired.
 * 
 * @author Alec Roelke
 */
public enum UpdateFrequency
{
    /** Only update on major version changes. */
    MAJOR("Major version change"),
    /** Update on major or minor version changes. */
    MINOR("Minor version change"),
    /** Update on major, minor, and revision changes. */
    REVISION("Revision change"),
    /** Update on all changes. */
    DAILY("Price update (daily)"),
    /** Don't update. */
    NEVER("Never");
    
    /** String label for this UpdateFrequency. */
    private final String label;

    /**
     * Create a new UpdateFrequency with the specified label.
     *
     * @param n label of the update frequency
     */
    UpdateFrequency(String n)
    {
        label = n;
    }

    @Override
    public String toString()
    {
        return label;
    }
}

File: src/main/scala/editor/database/version/UpdateFrequency.scala
package editor.database.version

enum UpdateFrequency(override val toString: String, val name: String) {
  case Major    extends UpdateFrequency("Major version change", "major")
  case Minor    extends UpdateFrequency("Minor version change", "minor")
  case Revision extends UpdateFrequency("Revision change", "revision")
  case Daily    extends UpdateFrequency("Daily", "daily")
  case Never    extends UpdateFrequency("Never", "never")
}